setup.py

```diff
@@ -29,58 +29,141 @@ def parse_requirements(filename):
-        'Intended Audience :: Developers',
-        'Topic :: Software Development :: Build Tools'
-        ],
-
-    license='Apache 2.0',
-
-    keywords='CMake build c++ fortran cross-platform cross-compilation',
-
-    extras_require={"test": test_requirements},
-)
+try:  # noqa: C901
+    setup(
+        name='cmake',
+
+        version=versioneer.get_version(),
+        cmdclass=versioneer.get_cmdclass(),
+
+        author='Jean-Christophe Fillion-Robin',
+        author_email='jchris.fillionr@kitware.com',
+
+        package_dir={'': 'src'},
+        packages=['cmake'],
+
+        cmake_install_dir='src/cmake/data',
+
+        entry_points={
+            'console_scripts': [
+                'cmake=cmake:cmake', 'cpack=cmake:cpack', 'ctest=cmake:ctest'
+            ]
+        },
+
+        url='https://cmake.org/',
+        download_url='https://cmake.org/download',
+        project_urls={
+            "Documentation": "https://cmake-python-distributions.readthedocs.io/",
+            "Source Code": "https://github.com/scikit-build/cmake-python-distributions",
+            "Mailing list": "https://groups.google.com/forum/#!forum/scikit-build",
+            "Bug Tracker": "https://github.com/scikit-build/cmake-python-distributions/issues",
+        },
+
+
+        description='CMake is an open-source, cross-platform family of '
+                    'tools designed to build, test and package software',
+
+        long_description=readme + '\n\n' + history,
+        long_description_content_type='text/x-rst',
+
+        classifiers=[
+            'License :: OSI Approved :: Apache Software License',
+            'License :: OSI Approved :: BSD License',
+            'Programming Language :: C',
+            'Programming Language :: C++',
+            'Programming Language :: Fortran',
+            'Programming Language :: Python',
+            'Operating System :: OS Independent',
+            'Development Status :: 5 - Production/Stable',
+            'Intended Audience :: Developers',
+            'Topic :: Software Development :: Build Tools'
+            ],
+
+        license='Apache 2.0',
+
+        keywords='CMake build c++ fortran cross-platform cross-compilation',
+
+        extras_require={"test": test_requirements},
+    )
+except BaseException:
+    # Note: This is a bare exception that re-raises so that we don't interfere
+    # with anything the installation machinery might want to do. Because we
+    # print this for any exception this msg can appear (e.g. in verbose logs)
+    # even if there's no failure. For example, SetupRequirementsError is raised
+    # during PEP517 building and prints this text. setuptools raises SystemExit
+    # when compilation fails right now, but it's possible this isn't stable
+    # or a public API commitment so we'll remain ultra conservative.
+    import platform
+    import subprocess
+    try:
+        import pkg_resources
+    except ImportError:
+        pass
```

Comment:
If this fails to import, then isn't the below code invalid, since it uses this?

Reply:
Right, I did not check too much this part of the code that came from cryptography and focused on correct glibc version being reported.
The easiest would be to replace `pkg_resources.DistributionNotFound` by a `BaseException` below. I'll do that.